### PR TITLE
docs: 📝 Introduce mkdocs-material into python-multipart project as new documentation 

### PR DESCRIPTION
**Pull request by onuralpszr**
As part of change process I am will be update every step for easy reviews and move parts to mkdocs-material and create config plus github action and It will replace new old documentation 

- [X] docs: ✨ decoders,exceptions,helper-functions,main-class,parsers,support-classes added
- [X] docs: ✨ set sphinx doc style as default
- [X] docs: ✨ initial mkdocs pages and documentation added, old docs removed
- [X] docs: ✨ mkdocs-material packages added as doc-dep
- [X] docs: 🧹 removal of old documents,packages,requirements
- [X] chore: 🧹 python.gitignore added
- [X] fix: 🐞 package typo fix
- [X] docs: ✨ home page, quickstart sections added

### How to Test

```console
git clone git@github.com:onuralpszr/python-multipart.git -b doc/mkdocs-material python-multipart-docs
cd python-multipart-docs
python3.10 -m venv .venv
source .venv/bin/activate 
pip install mkdocs==1.5.3 mkdocs-material==9.5.9 mkdocs-git-authors-plugin==0.7.2 mkdocs-git-revision-date-localized-plugin==1.2.4 mkdocs-material-extensions==1.3.1 pymdown-extensions==10.7 mkdocstrings==0.24.0 mkdocstrings-python==1.8.0 mkdocs-autorefs==0.5.0 mkdocs-git-committers-plugin-2==2.2.3
mkdocs serve
```

Fix : #101

### CODE REVIEW

**Comment by onuralpszr** [comment]
Hello @Kludex I am trying to replicate old document as it is as possible reference for "page structure" and as preview I had to change document section so it can render properly 

Light theme

image

Dark theme

image


If you have an different idea or suggestion also please let me know. Until then I will follow based on current documentation structure as possible as I can after initial work done, we can merge or change in here and then merge but I want to give you update how is it look in early.

**Comment by onuralpszr** [comment]
### How to Test

```console
git clone git@github.com:onuralpszr/python-multipart.git -b doc/mkdocs-material python-multipart-docs
cd python-multipart-docs
python3.10 -m venv .venv
source .venv/bin/activate 
pip install mkdocs==1.5.3 mkdocs-material==9.5.9 mkdocs-git-authors-plugin==0.7.2 mkdocs-git-revision-date-localized-plugin==1.2.4 mkdocs-material-extensions==1.3.1 pymdown-extensions==10.7 mkdocstrings==0.24.0 mkdocstrings-python==1.8.0 mkdocs-autorefs==0.5.0 mkdocs-git-committers-plugin-2==2.2.3
mkdocs serve
```

**Comment by onuralpszr** [comment]
@Kludex  it is ready for review and how to test also added for quickly copy/paste and see it. It should be mostly ready for first round I believe so please check it when you had time

Thank you.  :)

**Comment by Kludex** [comment]
I was fixing a lot of stuff, and now you pushed more stuff :/

**Comment by onuralpszr** [comment]
> I was fixing a lot of stuff, and now you pushed more stuff :/

I did not know, I am stopping for now.

**Comment by Kludex** [comment]
I'm force pushing my changes, FYI. (the 3 last commits will be lost)

File: .gitignore [hunk 1]
```diff
@@ -1,29 +1,6 @@
...
-*.py[cod]
-
-# C extensions
-*.so
-
-# Packages
-*.egg
-*.egg-info
-MANIFEST
-dist
```
Comment (Kludex) [inline .gitignore:15]: dist needs to be ignored

**Review by Kludex — APPROVED** [review]
Thanks! 👍 

Further improvements are welcome, but I prefer to go with the simplest, so I've removed a bit of complexity.

**Comment by onuralpszr** [comment]
> Thanks! 👍
> 
> Further improvements are welcome, but I prefer to go with the simplest, so I've removed a bit of complexity.

Okay thank you.  I understand, since this was first it is okay.

**Pull request merged**